File: AHome/19.12/BallThreadClass/Ball/Ball.cs
using System;
using System.Drawing;
using System.Threading;

namespace Ball
{
    public class Ball
    {
        public int posX;
        public int posY;

        public Color color;
        public int width;
        public int dx;
        public int dy;

        private Random rnd;

        public Ball(int x, int y)
        {
            posX = x;
            posY = y;

            rnd = new Random();
            color = Color.DarkSlateBlue;
            width = rnd.Next(40, 70);
            dx = rnd.Next(-8, 8);
            dy = rnd.Next(-8, 8);
        }

        public void DrawBall(Frame frame)
        {
            Graphics gr = frame.CreateGraphics();
            gr.FillEllipse(new SolidBrush(color), posX, posY, width, width);
        }
        public void EraseBall(Frame frame)
        {
            Graphics gr = frame.CreateGraphics();
            gr.FillEllipse(new SolidBrush(frame.BackColor), posX, posY, width, width);
        }
        public void MoveBall(Frame frame)
        {
            EraseBall(frame);
            posX += dx;
            posY += dy;

            if (posX <= 0 || posX + width >= frame.Width)
            {
                dx = -dx;
                width -= 2;
                color = Color.FromArgb(rnd.Next(200, 255), rnd.Next(0, 100), rnd.Next(0, 100));
            }
            if (posY <= 0 || posY + width >= frame.Height)
            {
                dy = -dy;
                width -= 2;
                color = Color.FromArgb(rnd.Next(0, 50), rnd.Next(0, 50), rnd.Next(200, 255));
            }
            DrawBall(frame);
        }

        public void RunBall(Frame frame)
        { 
            while (true)
            {
                Thread.Sleep(50);
                MoveBall(frame);
            }
        }
    }
}

File: AHome/19.12/SimplePaint/app/src/main/java/com/example/ilya/simplepaint/Ball.java
package com.example.vladislav.simplepaint;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;

import java.util.Random;

public class Ball {
        public int width;
        public int posX;
        public int posY;

        public int dx;
        public int dy;

        private Random rnd;

        public Ball(int x, int y)
        {
            rnd = new Random();
            width = rnd.Next(30, 60);
            posX = x;
            posY = y;
            dx = rnd.Next(-10, 10);
            dy = rnd.Next(-10, 10);
        }

        public void MoveBubble(Canvas can)
        {
            ball.posX += ball.dx;
            ball.posY += ball.dy;

            if (ball.posX <= 0 || ball.posX + ball.width >= ClientSize.Width)
            {
                ball.dx = -ball.dx;
            }
            if (ball.posY <= 0 || ball.posY + ball.width >= ClientSize.Height)
            {
                ball.dy = -ball.dy;
            }
        }
    }